flink-core/src/test/java/org/apache/flink/api/java/typeutils/PojoParametrizedTypeExtractionTest.java

```diff
@@ -0,0 +1,101 @@
+package org.apache.flink.api.java.typeutils;
+
+import org.apache.flink.api.common.functions.MapFunction;
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.Serializable;
+
+/**
+ *  Tests concerning type extraction of Parametrized Pojo and its superclasses.
+ */
+public class PojoParametrizedTypeExtractionTest {
+
+	@Test
+	public void testCreateTypeInfo(){
+		/// Init
+		final TypeInformation<ParametrizedParentImpl> directTypeInfo = TypeExtractor.createTypeInfo(ParametrizedParentImpl.class);
+
+		/// Check
+		Assert.assertTrue(directTypeInfo instanceof PojoTypeInfo);
+	}
+
+	@Test
+	public void testFieldExtraction(){
+		/// Init
+		final TypeInformation<ParametrizedParentImpl> directTypeInfo = TypeExtractor.createTypeInfo(ParametrizedParentImpl.class);
+
+		// Extract
+		TypeInformation<?> directPojoFieldTypeInfo = ((PojoTypeInfo) directTypeInfo)
+			.getPojoFieldAt(0)
+			.getTypeInformation();
+
+		// Check
+		Assert.assertTrue(directPojoFieldTypeInfo instanceof PojoTypeInfo);
+	}
+
+	@Test
+	public void testMapReturnTypeInfo(){
+		/// Init
+		final TypeInformation<ParametrizedParentImpl> directTypeInfo = TypeExtractor.createTypeInfo(ParametrizedParentImpl.class);
+
+		// Extract
+		TypeInformation<ParametrizedParentImpl> mapReturnTypeInfo = TypeExtractor
+			.getMapReturnTypes(new ConcreteMapFunction(), directTypeInfo);
+
+		// Check
+		Assert.assertTrue(mapReturnTypeInfo instanceof PojoTypeInfo);
+		Assert.assertEquals(directTypeInfo, mapReturnTypeInfo);
```

Comment:
I tend to use an explicit `PojoTypeInfo` as the expected result (namely we create a PojoTypeInfo directly). Since both the two type extraction logic are based on similar logic, it is possible that some bad implementation may cause the same errors for the two processes.

Besides, if we use explicit expected result, we could also remove the previous `instanceof` assert.